src/System/extract_state.rs

```diff
@@ -27,11 +27,12 @@ pub fn get_state(
                 _ => unreachable!(),
             };
         }
-
-        Ok(State::create(
+        let mut state = State::create(
             build_location_tuple(&mut locations.iter(), machine, system)?,
```

Comment:
Here we construct a `LocationTuple`:
1) Should we rename it to `LocationTree` as in the protobuf and,
2) Should the name of the function be `build_`..., which could imply the builder pattern or am I thinking about it to much?

This may be an issue throughout Reveaal, so maybe we should jsut make a seperate issue for it and let it go for now?

Reply:
I think separate issue yeah, but I agree. There are like 30+ `build_*` functions and none of them are builder patterns 😅We also use `create_*` often. I don't know what prefix (if any) we should use in general, but let's discuss in the issue.

Reply:
I have renamed everything to LocationTree now. Also variables so we don't reference tuples anymore 🙂